wade-review.md

```diff
@@ -0,0 +1,596 @@
+
+    private KeyValueStore<String, GitHubPRStateCounter> kvStore;
+                @Override
+                public void init(final ProcessorContext<String, GitHubPRStateCounter> context) {
+                    this.kvStore = context.getStateStore(STORE_NAME);
+
+                    context.schedule(Duration.ofSeconds(1), PunctuationType.STREAM_TIME, timestamp -> {
+                        GitHubPRStateCounter entry = kvStore.get(STORE_KEY);
+                        System.out.printf("Store value %s%n", entry);
+                        context.forward(new Record<>("pr", entry, timestamp));
+                    });
+	}
+              @Override
+                public void process(final Record<String, GitHubPRInfo> record) {
+
+                }
+            };
+        }
+}
+```
+
+Next, I’ll write the process method to  take in the events, mark them as open or closed, increment the count, and stash them in the state store.
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> { 
+	static final String STORE_KEY = "state-counter";
+   @Override
+   public Processor<String, GitHubPRInfo, String, GitHubPRStateCounter> get() {
+		return new Processor<>() {
+
+    private KeyValueStore<String, GitHubPRStateCounter> kvStore;
+                @Override
+                public void init(final ProcessorContext<String, GitHubPRStateCounter> context) {
+                    this.kvStore = context.getStateStore(STORE_NAME);
+
+                    context.schedule(Duration.ofSeconds(1), PunctuationType.STREAM_TIME, timestamp -> {
+                        GitHubPRStateCounter entry = kvStore.get(STORE_KEY);
+                        System.out.printf("Store value %s%n", entry);
+                        context.forward(new Record<>("pr", entry, timestamp));
+                    });
+	}
+              @Override
+                public void process(final Record<String, GitHubPRInfo> record) {
+                    GitHubPRInfo prInfo = record.value();
+
+                }
+            };
+        }
+}
+```
+We’ll get the value of the record and start a chain of logic. 
+
+```java
+SHOULD I ZOOM IN HERE I WONDER
+                    if(!prInfo.state().equals(NO_STATE)) {
+                        GitHubPRStateCounter stateCounter = kvStore.get(STORE_KEY);
+                        if (stateCounter == null) {
+                            stateCounter = new GitHubPRStateCounter();
+                        }
+                        if (prInfo.state().equalsIgnoreCase("open")) {
+                            stateCounter.setOpen(stateCounter.getOpen() + 1);
+                        } else {
+                            stateCounter.setClosed(stateCounter.getClosed() + 1);
+                        }
+                        kvStore.put(STORE_KEY, stateCounter);
+                    }
+```
+
+If there’s no state and the state counter is null, we instantiate a new state counter. 
```

Comment:
Reveal the code section by section. So start by just revealing the null check and leave the other bits out. Then as you explain, reveal each individual piece.

Reply:
https://github.com/Cerchie/notes-for-gh-video/pull/2/commits/10b6005a6176a000f4ffe953389e26157ca87d81